libs/clients/fiskistofa/src/lib/api.ts

```diff
@@ -0,0 +1,227 @@
+import { AuthHeaderMiddleware } from '@island.is/auth-nest-tools'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import { Inject, Injectable } from '@nestjs/common'
+import { ConfigType } from '@nestjs/config'
+import {
+  Configuration,
+  SkipApi,
+  StadaSkipsApi,
+  V1SkipHeitiHeitiGetRequest,
+  V1SkipSkipnumerGetRequest,
+  V1StadaskipsKvotategundirAlmanaksarArGetRequest,
+  V1StadaskipsKvotategundirFiskveidiarFiskveidiarGetRequest,
+  V1StadaskipsSkipnumerAlmanaksarArDeilistofnarBreyttPostRequest,
+  V1StadaskipsSkipnumerAlmanaksarArDeilistofnarGetRequest,
+  V1StadaskipsSkipnumerFiskveidiarFiskveidiarBreyttPostRequest,
+  V1StadaskipsSkipnumerFiskveidiarFiskveidiarGetRequest,
+  V1StadaskipsSkipnumerFiskveidiarFiskveidiarKvotiBreyttPostRequest,
+} from '../../gen/fetch'
+import { FiskistofaClientConfig } from './fiskistofaClient.config'
+import {
+  mapShipStatusForTimePeriod,
+  mapShipStatusForCalendarYear,
+  mapQuotaType,
+} from './utils'
+import { FetchError } from '@island.is/clients/middlewares'
+import { LazyDuringDevScope } from '@island.is/nest/config'
+
+@Injectable({ scope: LazyDuringDevScope })
+export class FiskistofaApi {
+  private stadaSkipsApi: StadaSkipsApi | null = null
+  private skipApi: SkipApi | null = null
+
+  constructor(
+    @Inject(FiskistofaClientConfig.KEY)
+    private clientConfig: ConfigType<typeof FiskistofaClientConfig>,
+  ) {}
+
+  async initialize(ignoreCache = false) {
+    if (this.stadaSkipsApi && this.skipApi && !ignoreCache) {
+      return
+    }
+
+    const stadaSkipsApi = new StadaSkipsApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-fiskistofa',
+          ...this.clientConfig.fetch,
+        }),
+        basePath: this.clientConfig.url,
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      }),
+    )
+
+    const skipApi = new SkipApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-fiskistofa',
+          ...this.clientConfig.fetch,
+        }),
+        basePath: this.clientConfig.url,
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      }),
+    )
+
+    const accessTokenResponse = await fetch(
+      this.clientConfig.accessTokenServiceUrl,
+      {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          grant_type: 'client_credentials',
+          client_id: this.clientConfig.accessTokenServiceClientId,
+          client_secret: this.clientConfig.accessTokenServiceClientSecret,
+          audience: this.clientConfig.accessTokenServiceAudience,
+        }),
+      },
+    )
```

Comment:
It would be really nice to use EnhancedFetch autoAuth to simplify the token management. Were you unable to get that working?

Reply:
I pushed a change where I'm using the autoauth option (I added the audience since it's required by the service)